pennylane/drawer/draw.py

```diff
@@ -573,16 +573,7 @@ def wrapper(*args, **kwargs_qnode):
                 qnode.construct(args, kwargs_qnode)
                 if isinstance(qnode.device, qml.devices.Device):
                     program = qnode.transform_program
-                    if any(
-                        isinstance(op, qml.measurements.MidMeasureMP)
-                        for op in qnode.tape.operations
-                    ):
-                        tapes, _ = qml.defer_measurements(qnode.tape, device=qnode.device)
-                    else:
-                        tapes = [qnode.tape]
-
-                    tapes, _ = program(tapes)
-                    tape = tapes[0]
+                    [tape], _ = program([qnode.tape])

```

Comment:
We probably need to do this for `draw_mpl` as well.

Reply:
This is for `draw_mpl`. I updated it in `draw` in #4901.